src/gui/gui_graphics.hpp

```diff
@@ -166,15 +170,16 @@ struct text_data : public text_base_data {
 		return (text_base_data::flags & is_edit_mask) != 0;
 	}
 };
+static_assert(sizeof(text_data) == 4);
 
 struct image_data {
 	static constexpr uint8_t frame_mask = 0x7F;
 	static constexpr uint8_t is_mask_mask = 0x80;
 
-	float scale = 1.0f;
-	dcon::gfx_object_id gfx_object;
-
-	uint8_t flags = 0;
+	float scale = 1.0f; // 4bytes
+	dcon::gfx_object_id gfx_object; // 6bytes
+	uint8_t flags = 0; // 7bytes
+	uint8_t padding = 0; // 8bytes
```

Comment:
This doesn't need padding, because it is going into a union. You need to make sure that the union as a whole is memset to 0 before anything is assigned to it.